storage2/src/storage.rs

```diff
@@ -29,16 +64,129 @@ impl Storage {
+                    db.put_cf(nct_cf, "tct", &tct_data)
+                })
+            })
+            .unwrap()
+            .await??;
+
+        // 2. Write the JMT to RocksDB
+        // We use wrapping_add here so that we can write `new_version = 0` by
+        // overflowing `PRE_GENESIS_VERSION`.
+        let old_version = self.latest_version().await?.unwrap();
+        let new_version = old_version.wrapping_add(1);
+        tracing::trace!(old_version, new_version);
+        let span = Span::current();
+        tokio::task::Builder::new()
+            .name("Storage::write_node_batch")
+            .spawn_blocking(move || {
+                span.in_scope(|| {
+                    let snap = self.latest_snapshot.read().clone();
+                    let jmt = JellyfishMerkleTree::new(snap.as_ref());
+                    // Write the unwritten changes from the state to the JMT.
+                    let (jmt_root_hash, batch) = jmt.put_value_set(
+                        state
+                            .unwritten_changes
+                            .into_iter()
+                            .map(|x| (KeyHash::from(x.0), x.1))
+                            .collect(),
+                        new_version,
+                    )?;
+
+                    // Apply the JMT changes to the DB.
+                    self.write_node_batch(&batch.node_batch)?;
+                    tracing::trace!(?jmt_root_hash, "wrote node batch to backing store");
+
+                    // TODO: 3. write the index tables to RocksDB
+                    // not yet implemented because the API for vectorized keys is not yet designed
+
+                    // 4. update the snapshot
+                    // Now that we've successfully written the new nodes, update the version.
+                    let jmt_version = new_version;
+                    let snapshot = db.snapshot();
+                    // Obtain the write-lock for the latest snapshot, and replace it with the new snapshot.
+                    let mut guard = self.latest_snapshot.write();
+                    *guard = Arc::new(Snapshot::new(snapshot, jmt_version, db));
+                    // Drop the write-lock (this will happen implicitly anyways, but it's good to be explicit).
+                    drop(guard);
+                    anyhow::Result::<()>::Ok(())
+                })
+            })
+            .unwrap()
+            .await
+            .unwrap();
+
+        Ok(())
     }
 }
+
+impl TreeWriter for Storage {
+    /// Writes a node batch into storage.
+    //TODO: Change JMT traits to accept owned NodeBatch
+    fn write_node_batch(&self, node_batch: &NodeBatch) -> Result<()> {
+        let db = self.db;
+        let node_batch = node_batch.clone();
+
+        // The writes have to happen on a separate spawn_blocking task, but we
+        // want tracing events to occur in the context of the current span, so
+        // propagate it explicitly:
+        let span = Span::current();
```

Comment:
I don't see this span getting used. Is this an old piece of code and comment, or is the span not getting propagated as it is intended to be?